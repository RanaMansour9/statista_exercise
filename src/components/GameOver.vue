<template>
  <div class="fixed inset-0 z-40">
    <div class="flex items-center justify-center w-full h-full bg-gray-500 opacity-95">
        <div class="flex flex-col items-center">
            <p class="flex items-center justify-center mb-12 text-2xl"><span class="mr-2 font-medium text-green-600"> {{ winnerName }} </span> is this GAME WINNER !!</p>
            <div class="flex">
                <button class="p-5 text-white rounded-md bg-emerald-600 w-36" @click="handleClick()">Play Again</button>
            </div>
        </div>
    </div>
  </div>
</template>


<script>
export default {
name: 'GameOver',
components: {
},
data () {
    return {
    }
},
props: {
    winnerName: {
    type: String,
    default: null
    }
},
computed: {
},
methods: {
    handleClick () {
        this.$emit('restartGame')
    }
},
unmounted () {
    this.$emit('resetWinnerName')
}

}
</script>



<style scoped>
</style>
