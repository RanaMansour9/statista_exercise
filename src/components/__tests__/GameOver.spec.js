import { it, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import GameOver from '../GameOver.vue'


const wrapper = shallowMount(GameOver)

it("Test GameOver component props", async () => {
    expect(wrapper.text()).toMatch('is this GAME WINNER !!')
});

it("Test if data is a function", () => {
    expect(typeof GameOver.data).toBe("function");
});

it('Snapshot track of changes UI testing', () => {
    expect(wrapper.text()).toMatchSnapshot()
})

it('Check winnerName', () => {
    expect(wrapper.vm.winnerName).toEqual(null)
})

it("Play Again button clicked", async () => {
    await wrapper.get("button").trigger("click")
    wrapper.vm.handleClick()
    expect(wrapper.emitted('restartGame')).toBeTruthy()
 })
