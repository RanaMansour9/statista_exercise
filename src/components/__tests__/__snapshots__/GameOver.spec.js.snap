// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot track of changes UI testing 1`] = `"is this GAME WINNER !!Play Again"`;

exports[`snapshot track of changes UI testing 1`] = `"is this GAME WINNER !!Play Again"`;
